influent-transport/src/main/java/influent/internal/util/InetNetwork.java

```diff
@@ -0,0 +1,47 @@
+package influent.internal.util;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+public interface InetNetwork {
+  static InetNetwork getBySpec(String spec) {
+    String[] addressComponents = spec.split("/", 2);
+    if (addressComponents.length != 2) {
+      throw new IllegalArgumentException("Invalid network: " + spec);
+    }
+    try {
+      int netmask = Integer.parseInt(addressComponents[1]);
+      InetAddress base = InetAddress.getByName(addressComponents[0]);
+      if (base.getAddress().length == 16) {
+        return new Inet6Network(base, netmask);
+      } else {
+        return new Inet4Network(base, netmask);
+      }
+    } catch (NumberFormatException | UnknownHostException e) {
+      throw new IllegalArgumentException("Invalid network: " + spec);
```

Comment:
Most Exception can accept an Exception.
This is helpful since a user can trace causes.

```java
throw new IllegalArgumentException("message", e);
```

Reply:
Thank you for comment. I will add original exception here.